src/NumericalAlgorithms/LinearSolver/IterationId.hpp

```diff
@@ -0,0 +1,55 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class IterationId.
+
+#pragma once
+
+#include <cstddef>
+#include <functional>
+#include <iosfwd>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace LinearSolver {
+
+/*!
+ * \ingroup LinearSolverGroup
+ * \brief Identifies a step in the linear solver algorithm
+ */
+struct IterationId {
+  size_t step_number{0};
+
+  IterationId() = default;
+
+  explicit IterationId(const size_t initial_step_number) noexcept
```

Comment:
`initial_step_number` suggests the start of the run to me (presumably that is always step 0).  I believe I've seen an `_in` suffix used for this, but here I'd just leave the constructor out and let people use aggregate initialization.